src/jit/CMakeLists.txt

```diff
@@ -23,6 +23,7 @@ set( JIT_SOURCES
   bitset.cpp
   block.cpp
   codegencommon.cpp
+  codegenlinear.cpp
```

Comment:
I gave some consideration to this:
Why not in codegencommon.cpp?
- I think we want more, smaller source files rather than larger ones.
- I think we are better off not combining legacy and non-legacy in the same file, at least going forward. If anything, we should consider moving the ryujit-only methods from codegencommon.cpp to codegenlinear.cpp.

I don't think people will be confused by the name. If they are looking for a code generation method, they will be able to find it. codegenlinearcommon.cpp is unnecessarily long.